### DISCUSSION TITLE
YN0057 is not listed in the error codes?

### DISCUSSION THREAD

**Discussion by SampsonCrowley** [yarnpkg/berry#3501] [Q&A] [answered]
[https://yarnpkg.com/advanced/error-codes](https://yarnpkg.com/advanced/error-codes)

Trying to migrate to V2 with workspace I get:

```bash
YN0057: │ @sampsonjs/stimulated: Invalid dependency range for '@sampsonjs/helpers'
```

no matter what dependency range I choose: `workspace:*`, `workspace:^`, exact version (`workspace:1.0.0-alpha.3`)

This is only a problem with PeerDependencies.

I think I should open an issue, but my app is already decently complex and I don't have a lot of free time to create a M.R.E.

Does anyone else have problems trying to use workspaces with peerDependencies?

FYI, I **am** installing the same dependency in `devDependencies` and that should satisfy the dependency range I would think

**Comment by arcanis** [comment, marked as answer]
Peer dependencies only support the workspace protocol starting from master. Use `yarn set version from sources`, or use `*` instead.

**Reply by merceyz** [reply]
Alternatively you can use the latest canary build
```sh
yarn set version canary
```

**Reply by SampsonCrowley** [reply]
Thanks
